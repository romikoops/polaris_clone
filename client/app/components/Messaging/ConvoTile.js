import React, { Component } from 'react';
import styles from './Messaging.scss';
// import { moment } from '../../constants';
export class ConvoTile extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const  { theme, conversation, viewConvo, convoKey } = this.props;
        console.log(theme);
        return (
            <div className={`flex-100 layout-row layout-align-start-start  ${styles.convo_tile_wrapper}`} onClick={() => viewConvo(conversation)}>
                <div className={`flex layout-row layout-align-center-start pointy layout-wrap  ${styles.convo_tile}`}>
                    <div className="flex-95 layout-row layout-align-start-center">
                        <p className="flex-none">Shipment: {convoKey}</p>
                    </div>
                    {/* <div className="flex-100 layout-row layout-align-start-center">
                        <p className="flex-none">Last Updated: {moment.unix(conversation.messages[0].timestamp).format('lll')}</p>
                    </div>*/}
                 </div>
            </div>
        );
    }
}
