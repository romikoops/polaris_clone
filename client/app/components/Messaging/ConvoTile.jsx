import React from 'react'
import styled from 'styled-components'
import styles from './Messaging.scss'
// import { moment } from '../../constants';
import PropTypes from '../../prop-types'

export function ConvoTile ({
  theme, conversation, viewConvo, convoKey
}) {
  const ConvoTileDiv = styled.div`
            
            box-shadow: 0 1px 1px 0 rgba(12,13,14,0.75);
            background: #fff;
    /*        border-width: 2px;
            border-style: solid;
            -webkit-border-image: 
              -webkit-gradient(linear, 0 0, 0 100%, from(${theme.colors.primary}), to(${
  theme.colors.secondary
})) 1 100%;
            -webkit-border-image: 
              -webkit-linear-gradient(${theme.colors.primary}, ${theme.colors.secondary}) 1 100%;
            -moz-border-image:
              -moz-linear-gradient(${theme.colors.primary}, ${theme.colors.secondary}) 1 100%;    
            -o-border-image:
              -o-linear-gradient(${theme.colors.primary}, ${theme.colors.secondary}) 1 100%;
            border-image:
              linear-gradient(to bottom, ${theme.colors.primary}, ${
  theme.colors.secondary
}) 1 100%*/
            border-radius: 10px;
        `
  const iconStyle = {
    background:
      theme && theme.colors
        ? `-webkit-linear-gradient(left, ${theme.colors.primary},${theme.colors.secondary})`
        : 'black'
  }
  return (
    <div
      className={`flex-100 layout-row layout-align-start-start  ${styles.convo_tile_wrapper}`}
      onClick={() => viewConvo(conversation)}
    >
      <ConvoTileDiv
        className={`flex layout-row layout-align-center-start pointy layout-wrap  ${
          styles.convo_tile
        }`}
      >
        <div className="flex-95 layout-row layout-align-start-center">
          <div className="flex-15-layout-row-layout-align-cetner-center">
            <i className="flex-none clip fa fa-ship" style={iconStyle} />
          </div>
          <p className="flex-none">Shipment: {convoKey}</p>
        </div>
      </ConvoTileDiv>
    </div>
  )
}

ConvoTile.propTypes = {
  theme: PropTypes.theme,
  viewConvo: PropTypes.func.isRequired,
  convoKey: PropTypes.string.isRequired,
  conversation: PropTypes.shape({
    messages: PropTypes.array
  }).isRequired
}

ConvoTile.defaultProps = {
  theme: null
}

export default ConvoTile
