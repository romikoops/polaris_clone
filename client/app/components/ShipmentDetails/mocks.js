// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
