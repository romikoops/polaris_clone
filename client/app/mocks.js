import { set, cloneDeep } from 'lodash'

/**
 * Checks if variable is an non-empty object
 *
 * True for `{a:1}`
 * Fase for `null, {}, [], [1,2,3]`
 */
const isObject = (x) => {
  const ok = x !== null && !Array.isArray(x) && typeof x === 'object'
  if (!ok) {
    return false
  }

  return Object.keys(x).length > 0
}
/* eslint-disable */
/**
 * Used in unit test to modify specific properties
 * with minimal lines of code
 *
 * You can specify `path` to be empty string
 * if the change affects more than one of object's branches
 */
export const change = (origin, pathRaw, rules) => {
  const willReturn = cloneDeep(origin)
  
  if (!isObject(rules)) {
    set(willReturn, pathRaw, rules)
    return willReturn
  }
  const path = pathRaw === '' ? '' : `${pathRaw}.`
  for (const ruleKey of Object.keys(rules)) {
    const rule = rules[ruleKey]
    if (!isObject(rule)) {
      set(
        willReturn,
        `${path}${ruleKey}`,
        rule
      )
      continue
    }
    Object.keys(rule).filter(subruleKey => !isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      set(
        willReturn,
        `${path}${ruleKey}.${subruleKey}`,
        subrule
      )
    })
    Object.keys(rule).filter(subruleKey => isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      Object.keys(subrule).map((deepKey) => {
        const deep = rule[subruleKey][deepKey]
        set(
          willReturn,
          `${path}${ruleKey}.${subruleKey}.${deepKey}`,
          deep
        )
      })
    })
  }

  return willReturn
}
/* eslint-enable */
export const identity = input => input

export const theme = {
  colors: {
    primary: '#333',
    secondary: '#fafafa'
  }
}

export const user = {
  role: { name: 'shipper' },
  company_name: 'FOO_COMPANY',
  currency: 'EUR',
  email: 'foo@bar.baz',
  first_name: 'John',
  guest: false,
  last_name: 'Doe',
  phone: '6345789'
}

export const history = {
  push: identity
}

export const tenant = {
  data: {
    id: 123,
    scope: {
      modes_of_transport: {
        ocean: {
          OCEAN_LOAD_TYPE: true
        },
        air: {},
        truck: {},
        rail: {}
      }
    },
    theme,
    subdomain: 'foosubdomain'
  }
}

export const req = {
  schedule: [{}],
  total: 345,
  shipment: {}
}

const schedulesInShipmentData = [
  { hub_route_key: 'OCEAN_HUB_ROUTE_KEY', mode_of_transport: 'ocean' },
  { hub_route_key: 'AIR_HUB_ROUTE_KEY', mode_of_transport: 'air' },
  { hub_route_key: 'TRUCK_HUB_ROUTE_KEY', mode_of_transport: 'truck' },
  { hub_route_key: 'RAIL_HUB_ROUTE_KEY', mode_of_transport: 'rail' }
]

export const shipmentInShipmentData = {
  load_type: 'OCEAN_LOAD_TYPE',
  selected_offer: { cargo: {}, total: { value: 77 } },
  schedules_charges: {
    OCEAN_HUB_ROUTE_KEY: { total: 40 },
    AIR_HUB_ROUTE_KEY: { total: 200 },
    TRUCK_HUB_ROUTE_KEY: { total: 75 },
    RAIL_HUB_ROUTE_KEY: { total: 125 }
  },
  total_goods_value: {
    value: 15,
    currency: 'USD'
  },
  trucking: {
    pre_carriage: { trucking_time_in_seconds: 55 }
  },
  has_on_carriage: false,
  has_pre_carriage: false,
  notes: 'FOO_NOTES',
  cargo_notes: 'FOO_CARGO_NOTES',
  eori: 1234,
  incoterm_text: 'FOO_INCOTERM_TEXT',
  total_price: {
    value: 12,
    currency: 'USD'
  }
}

export const shipmentData = {
  contacts: [],
  shipment: shipmentInShipmentData,
  documents: [],
  cargoItems: [],
  containers: [],
  schedules: schedulesInShipmentData,
  locations: {
    startHub: 'FOO_START_HUB',
    endHub: 'FOO_END_HUB',
    destination: {
      street_number: 9,
      street: 'BAR_STREET',
      city: 'BAR_CITY',
      country: 'China',
      zip_code: 845321
    },
    origin: {
      street_number: 7,
      street: 'FOO_STREET',
      city: 'FOO_CITY',
      country: 'Germany',
      zip_code: 21177
    }
  }
}

export const match = {
  params: {}
}

export const shipments = {
  direction: 'export',
  finished: [],
  open: [],
  requested: []
}

export const shipment = {
  origin_hub: { name: 'FOO_ORIGIN_HUB' },
  destination_hub: { name: 'FOO_DESTINATION_HUB' },
  id: 654,
  status: 'FOO_STATUS',
  clientName: 'FOO_CLIENT_NAME',
  planned_etd: 789,
  imc_reference: 'FOO_IMC_REFERENCE',
  schedule_set: []
}

export const location = {
  primary: true,
  id: 246,
  street_number: 579,
  streetNumber: 579,
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_STREET',
  city: 'FOO_CITY',
  zip_code: '22456',
  zipCode: '22456',
  country: 'Germany'
}

export const address = {
  id: 13579,
  street_number: '1',
  street: 'Uhlandweg',
  city: 'Hamburg',
  zip_code: '22848',
  country: 'Germany'
}

export const hub = {
  location,
  name: 'FOO_HUB_NAME'
}

export const client = {
  first_name: 'Sebastian',
  last_name: 'Muller',
  email: 'muller55@yahoo.de',
  company_name: 'Mehristweniger',
  password: 'pwd'
}

export const route = {
  id: 24680
}

export const vehicleType = {
  id: 827364,
  name: 'FOO_VEHICLE_TYPE_NAME',
  mode_of_transport: 'FOO_VEHICLE_TYPE_TRANSPORT'
}

export const schedule = {
  hub_route_key: 'FOO_SCHEDULE_ROUTE_KEY',
  id: 555777,
  eta: 441188
}

export const charge = {
  id: 654297
}

export const locations = {
  origin: {},
  destination: {}
}

class MapMock {
  constructor (x) {
    this.x = x
  }
  bindTo () {
    return this.x
  }
  setContent () {
    return this.x
  }
  addListener () {
    return this.x
  }
}

export const gMaps = {
  InfoWindow: MapMock,
  LatLngBounds: MapMock,
  Map: MapMock,
  MapTypeId: { ROADMAP: '' },
  Marker: MapMock,
  Point: MapMock,
  Size: MapMock,
  places: { Autocomplete: MapMock }
}
