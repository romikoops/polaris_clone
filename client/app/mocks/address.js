export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
