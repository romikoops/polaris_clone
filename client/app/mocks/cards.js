export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
