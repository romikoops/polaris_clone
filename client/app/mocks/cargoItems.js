export const firstCargoItem = {
  cargo_item_type_id: 'foo',
  chargeable_weight: 250,
  dangerous_goods: false,
  description: 'FOO_CARGO_ITEM_DESCRIPTION',
  dimension_x: 10,
  dimension_y: 60,
  dimension_z: 40,
  gross_weight: 76,
  hs_codes: [4],
  hs_text: 'FOO_CARGO_ITEM_HS_TEXT',
  id: 1,
  key: 'FOO_CARGO_ITEM_KEY',
  payload_in_kg: 200,
  quantity: 5,
  stackable: true,
  tare_weight: 20,
  weight: 220
}

export const secondCargoItem = {
  cargo_item_type_id: 'bar',
  chargeable_weight: 150,
  description: 'BAR_CARGO_ITEM_DESCRIPTION',
  dimension_x: 100,
  dimension_y: 50,
  dimension_z: 70,
  hs_codes: [],
  hs_text: 'BAR_CARGO_ITEM_HS_TEXT',
  id: 2,
  key: 'BAR_CARGO_ITEM_KEY',
  payload_in_kg: 100,
  quantity: 7,
  stackable: false
}

export const cargoItems = [firstCargoItem, secondCargoItem]
export const cargoItemTypes = { foo: { description: 'FOO_TYPE' }, bar: { description: 'BAR_TYPE' } }

export const cargoItemGroup = {
  dimension_y: parseFloat(firstCargoItem.dimension_y) * parseInt(firstCargoItem.quantity, 10),
  dimension_z: parseFloat(firstCargoItem.dimension_z) * parseInt(firstCargoItem.quantity, 10),
  dimension_x: parseFloat(firstCargoItem.dimension_x) * parseInt(firstCargoItem.quantity, 10),
  payload_in_kg: parseFloat(firstCargoItem.payload_in_kg) * parseInt(firstCargoItem.quantity, 10),
  quantity: 1,
  groupAlias: 1,
  cargo_group_id: firstCargoItem.id,
  chargeable_weight: parseFloat(firstCargoItem.chargeable_weight) * parseInt(firstCargoItem.quantity, 10),
  hsCodes: firstCargoItem.hs_codes,
  hsText: firstCargoItem.hs_text,
  cargoType: { description: 'FOO_TYPE' },
  volume:
    parseFloat(firstCargoItem.dimension_y) *
    parseFloat(firstCargoItem.dimension_x) *
    parseFloat(firstCargoItem.dimension_z) /
    1000000 *
    parseInt(firstCargoItem.quantity, 10),
  items: [firstCargoItem]
}