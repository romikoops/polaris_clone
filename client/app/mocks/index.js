export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
