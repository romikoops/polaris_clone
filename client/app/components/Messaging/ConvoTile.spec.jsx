import * as React from 'react'
import { mount, shallow } from 'enzyme'
import { theme, shipment, identity } from '../../mocks'

/**
 * ISSUE
 * `theme && theme.colors` is redundant safeguard
 * because lines before `theme.colors` is used without safeguard
 */

// eslint-disable-next-line
import { ConvoTile } from './ConvoTile'

const propsBase = {
  theme,
  viewConvo: identity,
  convoKey: 'FOO_KEY',
  conversation: {
    messages: ['FOO_MESSAGE', 'BAR_MESSAGE']
  },
  shipment
}

const createWrapper = propsInput => mount(<ConvoTile {...propsInput} />)

test('shallow render', () => {
  expect(shallow(<ConvoTile {...propsBase} />)).toMatchSnapshot()
})

test('shipment.status === requested', () => {
  const props = {
    ...propsBase,
    shipment: {
      ...shipment,
      status: 'requested'
    }
  }
  expect(shallow(<ConvoTile {...props} />)).toMatchSnapshot()
})

test('shipment.status === confirmed', () => {
  const props = {
    ...propsBase,
    shipment: {
      ...shipment,
      status: 'confirmed'
    }
  }
  expect(shallow(<ConvoTile {...props} />)).toMatchSnapshot()
})

test('shipment.convoKey is truthy', () => {
  const props = {
    ...propsBase,
    shipment: {
      ...shipment,
      convoKey: 'FOO_CONVO_KEY'
    }
  }
  expect(shallow(<ConvoTile {...props} />)).toMatchSnapshot()
})

test('click calls props.viewConvo', () => {
  const props = {
    ...propsBase,
    viewConvo: jest.fn()
  }
  const wrapper = createWrapper(props)
  const clickableDiv = wrapper.find('.convo_tile_wrapper').first()

  expect(props.viewConvo).not.toHaveBeenCalled()

  clickableDiv.simulate('click')

  expect(props.viewConvo).toHaveBeenCalled()
})
