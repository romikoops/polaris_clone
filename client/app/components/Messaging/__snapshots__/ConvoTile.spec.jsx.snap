// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convoKey is truthy 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start  convo_tile_wrapper"
  onClick={[Function]}
>
  <styled.div
    className="flex layout-row layout-align-center-start-space-between pointy layout-wrap convo_tile "
  >
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      <div
        className="flex-15-layout-row-layout-align-start-center"
      >
        <i
          className="flex-none clip fa fa-truck"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
      <div
        className="flex-5"
      />
      <p
        className="flex-none"
      >
        Shipment Reference
        :
      </p>
      <div
        className="flex-5"
      />
      <b>
        FOO_CONVO_KEY
      </b>
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      Rejected by Admin
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      STATUS
      :
      <div
        className="flex-5"
      />
      <i>
        <b>
          REJECTED
        </b>
      </i>
      <div
        className="flex-5"
      />
      <i
        className="fa fa-question"
        style={
          Object {
            "color": "#A19F9F",
          }
        }
      />
    </div>
    <p
      className="flex-none"
    />
  </styled.div>
</div>
`;

exports[`shallow render 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start  convo_tile_wrapper"
  onClick={[Function]}
>
  <styled.div
    className="flex layout-row layout-align-center-start-space-between pointy layout-wrap convo_tile "
  >
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      <div
        className="flex-15-layout-row-layout-align-start-center"
      >
        <i
          className="flex-none clip fa fa-truck"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
      <div
        className="flex-5"
      />
      <p
        className="flex-none"
      >
        Shipment Reference
        :
      </p>
      <div
        className="flex-5"
      />
      <b>
        FOO_KEY
      </b>
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      Rejected by Admin
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      STATUS
      :
      <div
        className="flex-5"
      />
      <i>
        <b>
          REJECTED
        </b>
      </i>
      <div
        className="flex-5"
      />
      <i
        className="fa fa-question"
        style={
          Object {
            "color": "#A19F9F",
          }
        }
      />
    </div>
    <p
      className="flex-none"
    />
  </styled.div>
</div>
`;

exports[`shipment.status === confirmed 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start  convo_tile_wrapper"
  onClick={[Function]}
>
  <styled.div
    className="flex layout-row layout-align-center-start-space-between pointy layout-wrap convo_tile "
  >
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      <div
        className="flex-15-layout-row-layout-align-start-center"
      >
        <i
          className="flex-none clip fa fa-truck"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
      <div
        className="flex-5"
      />
      <p
        className="flex-none"
      >
        Shipment Reference
        :
      </p>
      <div
        className="flex-5"
      />
      <b>
        FOO_KEY
      </b>
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      Rejected by Admin
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      STATUS
      :
      <div
        className="flex-5"
      />
      <i>
        <b>
          REJECTED
        </b>
      </i>
      <div
        className="flex-5"
      />
      <i
        className="fa fa-check"
        style={
          Object {
            "color": "#3ACE62",
          }
        }
      />
    </div>
    <p
      className="flex-none"
    />
  </styled.div>
</div>
`;

exports[`shipment.status === requested 1`] = `
<div
  className="flex-100 layout-row layout-align-start-start  convo_tile_wrapper"
  onClick={[Function]}
>
  <styled.div
    className="flex layout-row layout-align-center-start-space-between pointy layout-wrap convo_tile "
  >
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      <div
        className="flex-15-layout-row-layout-align-start-center"
      >
        <i
          className="flex-none clip fa fa-truck"
          style={
            Object {
              "background": "-webkit-linear-gradient(left, #333,#fafafa)",
            }
          }
        />
      </div>
      <div
        className="flex-5"
      />
      <p
        className="flex-none"
      >
        Shipment Reference
        :
      </p>
      <div
        className="flex-5"
      />
      <b>
        FOO_KEY
      </b>
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      Rejected by Admin
    </div>
    <div
      className="flex-95 layout-row layout-align-start-center"
    >
      STATUS
      :
      <div
        className="flex-5"
      />
      <i>
        <b>
          REJECTED
        </b>
      </i>
      <div
        className="flex-5"
      />
      <i
        className="fa fa-hourglass-half"
        style={
          Object {
            "color": "#DEBC4C",
          }
        }
      />
    </div>
    <p
      className="flex-none"
    />
  </styled.div>
</div>
`;
